packages/astro/src/@types/astro.ts

```diff
@@ -167,33 +198,24 @@ export interface ManifestData {
   routes: RouteData[];
 }
 
-export interface PaginatedCollectionProp<T = any> {
-  /** result */
-  data: T[];
-  /** metadata */
-  /** the count of the first item on the page, starting from 0 */
-  start: number;
-  /** the count of the last item on the page, starting from 0 */
-  end: number;
-  /** total number of results */
-  total: number;
-  /** the current page number, starting from 1 */
-  currentPage: number;
-  /** number of items per page (default: 25) */
-  size: number;
-  /** number of last page */
-  lastPage: number;
-  url: {
-    /** url of the current page */
-    current: string;
-    /** url of the previous page (if there is one) */
-    prev: string | undefined;
-    /** url of the next page (if there is one) */
-    next: string | undefined;
-  };
+/**
+ * paginate() Options
+ * Docs: https://docs.astro.build/guides/pagination/#calling-the-paginate-function
+ */
+export interface PaginateOptions {
+  /** the number of items per-page (default: `10`) */
+  pageSize?: number;
+  /** key: value object of page params (ex: `{ tag: 'javascript' }`) */
+  params?: Params;
+  /** object of props to forward to `page` result */
+  props?: Props;
 }
 
-export interface PaginatedCollectionResult<T = any> {
+/**
+ * Page Prop
+ * Docs: https://docs.astro.build/guides/pagination/#using-the-page-prop
+ */
+export interface Page<T = any> {
```

Comment:
One example of renaming `PaginatedCollectionProps` to simply `Page` for users to find more easily.